app/uk/gov/hmrc/pensionschemereturn/transformations/nonsipp/MemberPaymentsTransformer.scala

```diff
@@ -131,26 +130,26 @@ class MemberPaymentsTransformer @Inject()(
         MemberPayments(
           recordVersion = out.recordVersion,
           memberDetails = details,
-          employerContributionsDetails = SectionDetails(
-            made = out.employerContributionMade.boolean,
-            completed =
-              if (!out.employerContributionMade.boolean) true
-              else !out.memberDetails.forall(_.noOfContributions.isEmpty)
-          ),
+          employerContributionsDetails = (
+            out.employerContributionMade.map(_.boolean),
+            out.memberDetails.forall(_.noOfContributions.nonEmpty)
+          ) match {
+            case (None, _) => SectionDetails.notStarted
+            case (Some(false), _) => SectionDetails(made = false, completed = true)
+            case (Some(true), completed) => SectionDetails(made = true, completed = completed)
+          },
           transfersInCompleted = out.memberDetails.forall(_.noOfTransfersIn.nonEmpty),
           transfersOutCompleted = out.memberDetails.forall(_.noOfTransfersOut.nonEmpty),
-          unallocatedContribsMade = unapply(out.unallocatedContribsMade),
+          unallocatedContribsMade = out.unallocatedContribsMade.map(_.boolean),
           unallocatedContribAmount = out.unallocatedContribAmount,
-          memberContributionMade = unapply(out.memberContributionMade),
-          lumpSumReceived = unapply(out.lumpSumReceived),
-          pensionReceived = unapply(out.pensionReceived),
+          memberContributionMade = out.memberContributionMade,
+          lumpSumReceived = out.lumpSumReceived,
+          pensionReceived = out.pensionReceived,
           benefitsSurrenderedDetails = {
```

Comment:
Are the SectionDetails for the rest of the member payments (everything apart from empl. contr. and surrendered benefits) still missing here?

Reply:
The rest of them can be inferred quite easily 👍

Reply:
Which ticket would that belong to?